src/Fixer/Phpdoc/PhpdocArrayStyleFixer.php

```diff
@@ -147,7 +147,7 @@ private function fixType(string $type): string
     {
         if (self::STRATEGY_FROM_ARRAY_TO_LIST !== $this->configuration['strategy']) {
             do {
-                $type = Preg::replace('/(.+)\[\]/', 'array<$1>', $type, -1, $count);
+                $type = Preg::replace('/(.+)\[\](?=($|\>))/', 'array<$1>', $type, -1, $count);
```

Comment:
you should not need `(?=\>)` here - `>` means we are fixing something on owning type, not the inner type

Reply:
How to handle types like `int[][][][]` then? `TypeExpression::walkTypes` does not split `[][][][]`.

Reply:
I mean why `'/(.+)\[\h*\]$/'` regex is not enough.

Reply:
Done, thanks for the hint about replace callback.